openedx_events/event_bus/__init__.py

```diff
@@ -1 +1,117 @@
-"""Classes and utility functions for the event bus."""
+"""
+Classes and utility functions for the event bus.
+
+This module includes the entry point for the producer.
+
+API:
+
+- ``get_producer`` returns an ``EventBusProducer`` singleton that should be used for sending all events
+  to the Event Bus. The backing implementation is chosen via the Django setting ``EVENT_BUS_PRODUCER``.
+"""
+
+import warnings
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import NoReturn, Optional
+
+from django.conf import settings
+from django.dispatch import receiver
+from django.test.signals import setting_changed
+from django.utils.module_loading import import_string
+
+from openedx_events.tooling import OpenEdxPublicSignal
+
+
+def _try_load(*, setting_name: str, expected_class: type, default):
+    """
+    Load an instance of ``expected_class`` as indicated by ``setting_name``.
+
+    The setting points to a callable (function or class) that will fetch or create an
+    instance of the expected class. If the configuration is missing or invalid,
+    or the callable throws an exception or returns the wrong type, the default is
+    returned instead.
+
+    Arguments:
+        setting_name: Name of a Django setting containing a dotted module path, indicating a callable
+        expected_class: The callable must produce an instance of this class object (or a subclass)
+        default: Object to return if any part of the lookup or loading fails
+    """
+    constructor_path = getattr(settings, setting_name, None)
+    if constructor_path is None:
+        warnings.warn(f"Event Bus setting {setting_name} is missing; component will be inactive.", UserWarning)
```

Comment:
Ah thanks, yeah, for some reason I was remembering it as a required argument.

()It can be passed as either arg or kwarg here as `warn` doesn't specify positional-only or keyword-only args.)